### PR TITLE
Add support for SMS templates

### PR DESCRIPTION
**Pull request by cb372**
Decided not to refactor too heavily at this point, so there is a bit of duplicated code here and there. Will probably need to do some refactoring when we add a third channel. Especially the tests will start to become unwieldy as the number of combinations of channels explodes.

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/templates/model/template/processed/CommTemplate.scala [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
 /* A full template for a comm, which will include templates for one or more channels */
 case class CommTemplate[M[_]: Applicative](
-                               email: Option[M[EmailTemplate[Id]]]
+                               email: Option[M[EmailTemplate[Id]]],
+                               sms: Option[M[SMSTemplate[Id]]]
                              ) {
 
-  def aggregate: M[CommTemplate[Id]] = {
-    Apply[M].map(email.sequenceU)(e => CommTemplate[Id](e))
+  def validate: ErrorsOr[Unit] = {
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/comms/templates/model/template/processed/CommTemplate.scala]: Do you think this should also collect any errors from the requiredData function?

**Comment by fleetwoods** [comment]
Looks as expected, would be keen to go through ideas on simplifying, as I cannot see anything obvious....so expecting some functional magic from you :)

File: src/main/scala/com/ovoenergy/comms/templates/model/template/processed/CommTemplate.scala (hunk 1 above)
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/comms/templates/model/template/processed/CommTemplate.scala]: Maybe it is just that the function name is a bit misleading, as there is plenty of validation going on elsewhere, when building up the templates. And this is purely checking that there is at least one channel present.

Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/templates/model/template/processed/CommTemplate.scala]: Good point, I'll rename the method to something more specific

**Comment by cb372** [comment]
I don't really have any good ideas either, but I'm keen to try and crowbar in a [catamorphism](http://chrislambda.github.io/blog/2014/01/30/catamorphisms-in-15-minutes/)!

**Comment by cb372** [comment]
@fleetwoods One thing we can revisit is whether we need to support the case of a template that is valid (in the sense that all the appropriate files are on S3) but does not have valid required data.

If we drop support for that and say that a template must have valid required data in order to be a valid template, it would simplify the library a lot. Basically we would lose one layer of `ErrorsOr` wrapping. It would also simplify things for the user of the library, as we wouldn't need to do the double-checking that we currently do in a few places ("is it a valid template, and does it have a valid required data").

Currently we are checking required data before publishing templates ([here](https://github.com/ovotech/comms-template-manager/blob/master/app/templates/TemplateValidator.scala#L41-L44)), so it's not possible to publish one of these semi-valid templates anyway.

**Pull request merged**